src/Core/Manifest/ClassManifest.php

```diff
@@ -496,4 +509,30 @@ protected function coalesceDescendants($class)
             return array();
         }
     }
+
+    /**
+     * Verify that cached data is valid for a single item
+     *
+     * @param array $data
+     * @return bool
+     */
+    protected function validateItemCache($data)
+    {
+        foreach (['classes', 'interfaces', 'traits'] as $key) {
+            // Must be set
+            if (!isset($data[$key])) {
+                return false;
+            }
+            // and an array
+            if (!is_array($data[$key])) {
+                return false;
+            }
+            // No numeric keys
+            $numericKeys = array_filter(array_keys($data), 'is_numeric');
```

Comment:
We shouldn't have code that covers upgrades from unstable to unstable versions...